### ISSUE TITLE
Redir duplicating the url

### ISSUE THREAD

**Issue by p3lim** [mholt/caddy#883]
#### 1. What version of Caddy are you running (`caddy -version`)?
0.8.3 on Ubuntu 16.04 x64, hosted on DigitalOcean.com.

#### 2. What are you trying to do?
Redirect www to apex.

#### 3. What is your entire Caddyfile?
```text
example.com {
    tls cert@example.com

    proxy / http://user.github.io {
        proxy_header X-Real-IP {remote}
        proxy_header Host {host}
    }
}

www.example.com {
    redir https://example.com/
}
```

I'm proxying the apex to Github Pages, the above was the only way I could make it work (improvement suggestions appreciated).

#### 4. How did you run Caddy (give the full command and describe the execution environment)?
`/usr/local/bin/caddy -log stdout -agree -conf ~/Caddyfile`

#### 5. What did you expect to see?
When I browse to `www.example.com` I should get redirected to `https://example.com`.

#### 6. What did you see instead (give full error messages and/or log)?
I get redirected to `https://example.com/example.com`.

#### 7. How can someone who is starting from scratch reproduce this behavior as minimally as possible?

```
curl getcaddy.com | bash
```
Then use the Caddyfile provided above.

**Comment by squirly** [comment]
I think you want `redir https://example.com{uri}` in the `www.example.com` block.

**Comment by p3lim** [comment]
@squirly I tried that, had the same result.

**Comment by mholt** [comment]
@p3lim Using `redir https://example.com{uri}` will preserve the URI if `www.example.com/foo` is loaded, it will redirect with `/foo`; the way you have it now will strip the `/foo` when it redirects. Probably not what you intended, and that's what @squirly meant.

Anyway, I am not seeing the same thing when I try your Caddyfile. Can you provide more information to reproduce this, or a screencast of the behavior from start to finish, maybe?

**Comment by p3lim** [comment]
@mholt Don't have much more information to add, I was able to replicate it by setting up a new "droplet" on DigitalOcean using the latest Ubuntu version, booted it up, did updates/upgrades, installed Caddy and set it up as described, set up A-records for my apex domain on my DNS and that's pretty much it.

I set up a CNAME redirect to implement what I wanted on my end (so I somewhat solved *my* issue), but the behavior I saw with Caddy still remains an issue.

**Comment by mholt** [comment]
Hmm, what's an apex domain? Only results are showing stuff about AWS but you're running on DO.

**Comment by p3lim** [comment]
An _apex domain_ is a root domain that does not contain a subdomain part. Apex domains are also known as base, bare, naked, root apex, or zone apex domains. In DNS provider settings apex domains are sometimes symbolized by an `@` symbol. For example:

* `example.com` is an apex domain because it doesn't have any subdomain parts.
* Whereas `www.example.com` is not an apex domain because it contains the subdomain part `www`.

**Comment by mholt** [comment]
Thanks for the info! Didn't know that was the name. I always just called those SLDs (second-level-domains).

I can't replicate this, but what if you remove the trailing `/`, in other words: `redir https://example.com{uri}`

This is exactly the same thing we're doing on the Caddy website, and it works fine, for example: https://www.caddyserver.com/docs

Anyway, please try this and let me know if that helps!

**Comment by p3lim** [comment]
I tried that, as I specified above.

I did manage to make it work however, by forcing port 80 and disabling tls:

```
www.example.com:80 {
    tls off
    redir https://example.com{uri}
}
```

**Comment by mholt** [comment]
Oh right. Well that's strange, only `:80` or `tls off` should be required to turn off HTTPS, but not both. And still not finding myself having to do this for any of my websites. Was your browser's cache disabled (the dev tools can force disable the cache)?

I'll keep trying to repro in the meantime, but I'll have to close this once I give up. 😛

**Comment by p3lim** [comment]
I've set up Firefox for development (disabling the cache is one thing) and using Chrome for everything else, but my ISP seems to be interfering with things.  
I killed the power to my modem for 10 or so minutes, forcing them to assign me a new IP (really wish I could get static :cry:) and probably do some other things on their end, and that seems to have resolved the issue.

I reverted back to my original configuration, and it worked fine, although I will keep the `{uri}` part of the redir.

Sorry for all the trouble!
